### ISSUE TITLE
Update/overhaul Checkbox accessibiltiy testing 

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#7364]
Definition of done: 
- [ ] convert existing tests to React Testing Library
- [ ] add Accessibility Checker test
- [ ] determine and implement missing WCAG 2.1 tests
- [ ] determine and document manual testing practices
